mypyc/genops.py

```diff
@@ -2437,8 +2525,33 @@ def visit_var(self, o: Var) -> None:
     def visit_yield_from_expr(self, o: YieldFromExpr) -> Value:
         raise NotImplementedError
 
-    def visit_yield_expr(self, o: YieldExpr) -> Value:
-        raise NotImplementedError
+    def visit_yield_expr(self, expr: YieldExpr) -> Value:
+
+        self.goto_new_block()
```

Comment:
Why the new block?

Reply:
This is one of those areas where I might generate extraneous labels. I need to make a new block because if the previous instruction was a `yield` statement, then I have a finished block and can't add anything to it. By making a new block every time I visit a yield statement, I might make extraneous blocks but I can guarantee that I will be adding to non-finished blocks.